### PR TITLE
Issue 522: Added DbType 

### PR DESCRIPTION
**Pull request by ObikeDev**
**Issue #522**
Added new DbType to Parameter for changing the underlying database type when utilizing DatabaseTarget.

TargetConfigurationTests.DatabaseParameterTest was added to verify configuration of parameters.
DatabaseTargetTests.ParameterFacetTest was updated to verify setting of DbType.

### CODE REVIEW

**Comment by Cable321** [comment]
What needs to happen to push this feature through at this point? Happy to help.

**Comment by 304NotModified** [comment]
Thanks @ObikeDev  and @Cable321 

I'm currently on holiday.

It would help if we could test this, in and unit test or just by installing it.

File: src/NLog/Targets/DatabaseParameterInfo.cs [hunk 1]
```diff
@@ -31,11 +31,12 @@
 // THE POSSIBILITY OF SUCH DAMAGE.
 // 
 
-#if !SILVERLIGHT
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseParameterInfo.cs]: note: in sync with database target

File: src/NLog/Targets/DatabaseParameterInfo.cs [hunk 2]
```diff
@@ -98,6 +99,13 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='10' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the database parameter type.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(DbType.AnsiString)]
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseParameterInfo.cs]: I think this is inconsistent with

The default is NVarChar. (https://msdn.microsoft.com/en-us/library/system.data.sqlclient.sqlparameter.dbtype(v=vs.110).aspx)

Which is `String` IMO (see https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings)

Proposal: make nullable and only set if non-null in databasetarget

File: src/NLog/Targets/DatabaseTarget.cs [hunk 3]
```diff
@@ -815,6 +815,11 @@ private void AddParametersToCommand(IDbCommand command, IList<DatabaseParameterI
                     p.Scale = par.Scale;
                 }
 
+                if (par.DbType != DbType.AnsiString)
```
Comment (304NotModified) [inline src/NLog/Targets/DatabaseTarget.cs]: see proposal commit before

**Comment by 304NotModified** [comment]
I think this isn't backwardscompatible with the default, so please make it optional (nullable)

**Comment by 304NotModified** [comment]
While the change looks good, I don't think this works.

as the current values are `string`s, and there is no parse, I don't think the SqlParameter converts the string to e.g. a int.

We need a parse or a "rawvalue", see #2849

**Comment by 304NotModified** [comment]
This PR is superseded by #2899 and is expected to be released in NLog 4.6

**Pull request closed without merging**